Data Prepper's s3 source: visibility_duplication_protection (#5665)
opensearch-project/documentation-website: _data-prepper/pipelines/configuration/sources/s3.md

If your S3 objects or Amazon SQS queues do not use [AWS Key Management Service (AWS KMS)](https://aws.amazon.com/kms/), remove the `kms:Decrypt` permission.

> If you do not enable `visibility_duplication_protection`, you may also remove the `sqs:ChangeMessageVisibility` permission from the SQS queue access.

Reviewer: ```suggestion
If you do not enable `visibility_duplication_protection`, you can remove the `sqs:ChangeMessageVisibility` permission from the SQS queue's access.
```

`visibility_timeout` | No | Duration | The visibility timeout to apply to messages read from the Amazon SQS queue. This should be set to the amount of time that Data Prepper may take to read all the S3 objects in a batch. Default is `30s`.
`wait_time` | No | Duration | The amount of time to wait for long polling on the Amazon SQS API. Default is `20s`.
`poll_delay` | No | Duration | A delay placed between the reading and processing of a batch of Amazon SQS messages and making a subsequent request. Default is `0s`.

> `visibility_duplication_protection` | No | Boolean | If set to `true`, Data Prepper will attempt to avoid duplicate processing by extending the visibility timeout of SQS messages. Until the data reaches the sink, Data Prepper will regularly call `ChangeMessageVisibility` to avoid reading the S3 object again. You must grant permissions to `ChangeMessageVisibility` on the IAM role to use this feature. Default is `false`.

Reviewer: ```suggestion
`visibility_duplication_protection` | No | Boolean | If set to `true`, Data Prepper attempts to avoid duplicate processing by extending the visibility timeout of SQS messages. Until the data reaches the sink, Data Prepper will regularly call `ChangeMessageVisibility` to avoid reading the S3 object again. To use this feature, you need to grant permissions to `ChangeMessageVisibility` on the IAM role. Default is `false`.
```
